src/main/java/com/xeeva/navigation/MainNav.java

```
@@ -1,317 +1,339 @@
+	@FindBy(id = "txtUnitPrice")	private Textbox txtUnitPrice;
+	@FindBy(id = "btnSubmit")	private Button btnSubmit;
+	@FindBy(xpath = ".//*[@id='txtItem']")	private Textbox itemDescription;
+	@FindBy(xpath = "//span[@id='spanCartValue']")	private Element lblCartValue;
+
+
+	// Update Cart Items - for non price agreement cart Items verifications
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/input")
+	private Textbox txtItemQuantity;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[4]/select")
+	private Listbox lstItemunitOfMeasure;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[2]/input[1]")
+	private Textbox txtItemUnitPrice;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/a/div")
+	private Button btnAddToCart;
+	@FindBy(xpath = ".//*[@value='Save Cart'][contains(@onclick, 'javascript')]") private Button btnsaveCartPopUp;
+	@FindBy(xpath = "//*[@id='btnPopupCP']") private Button btnCheckOut;
+
+	/** Constructor **/
+
+	public MainNav(OrasiDriver driver) {
+		this.driver = driver;
+		pl.isDomComplete(driver);
+		ElementFactory.initElements(driver, this);
+	}
+
+	/** Page Interactions **/
+
+	public void pageLoaded() {
+		lnkLogout.syncVisible(10, false);
+	}
+
+	public boolean isLogoutDisplayed() {
+		try {
+			Thread.sleep(3000);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return lnkLogout.syncVisible(20, false);
+	}
+
+
+	/**
+	 * @summary  Method to click on Cart-Items Link
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+
+	public void clickCartItemsLink() {
+		driver.executeJavaScript("arguments[0].click();", lnkCartItem);
+		driver.setPageTimeout(4);
+	}
+
+	/**
+	 * @summary Method for Application Logout
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void SaveCart() {
+		if (btnSaveCart.syncVisible(1, false)){
+			btnSaveCart.click();
+		} else {
+			TestReporter.logStep("Cart Empty to Save");
```

If the save button is not able to be clicked - do we need to fail instead of just reporting  log statement?